docs/source/articles/implement-the-credit-card-tender-type.html.md

```diff
@@ -0,0 +1,298 @@
+Refer to the following sources:
+
+* [Workarea Core implementation](https://github.com/workarea-commerce/workarea/blob/v3.4.18/core/app/models/workarea/payment/store_credit_card.rb)
+* [Workarea Core test case](https://github.com/workarea-commerce/workarea/blob/v3.4.18/core/test/models/workarea/payment/store_credit_card_test.rb)
+* [Workarea Braintree implementation extensions](https://github.com/workarea-commerce/workarea-braintree/blob/v1.0.3/app/models/workarea/payment/store_credit_card.decorator)
+* [Workarea Braintree test case extensions](https://github.com/workarea-commerce/workarea-braintree/blob/v1.0.3/test/models/workarea/payment/store_credit_card_test.decorator)
+
+
+Operation Implementations
+----------------------------------------------------------------------
+
+Finally, to finish implementing the credit card tender type, you must implement each credit card operation: authorize, purchase, capture, and refund.
+In each case, decorate the operation implementation to account for differences between the default gateway and your chosen gateway.
+
+For the credit card tender type, the authorize and purchase operations must handle tokenization.
+There are two general patterns for this behavior.
+The first, as demonstrated in the default implementation, is to tokenize the card in a separate request before authorizing the transaction
+([source](https://github.com/workarea-commerce/workarea/blob/v3.4.18/core/app/models/workarea/payment/authorize/credit_card.rb)):
+
+```ruby
+def complete!
+  return unless StoreCreditCard.new(tender, options).save!
+
+  transaction.response = handle_active_merchant_errors do
+    gateway.authorize(
+      transaction.amount.cents,
+      tender.to_token_or_active_merchant,
+      transaction_options
+    )
+  end
+end
+```
+
+The alternative, preferred by Braintree and other gateways, is to tokenize and authorize in the same request, in which case you must save the token after a successful authorization
+([source](https://github.com/workarea-commerce/workarea-braintree/blob/v1.0.3/app/models/workarea/payment/authorize/credit_card.decorator)):
+
+```ruby
+def complete!
+  transaction.response = handle_active_merchant_errors do
+    if tender.token.present?
+      gateway.authorize(
+        transaction.amount.cents,
+        tender.token,
+        {
+          payment_method_token: true,
+          order_id: order_id,
+          email: email,
+          billing_address: billing_address
+        }
+      )
+    else
+      gateway.authorize(
+        transaction.amount.cents,
+        tender.to_active_merchant,
+        {
+          store: true,
+          order_id: order_id,
+          email: email,
+          billing_address: billing_address
+        }
+      )
+    end
+  end
+
+  if transaction.response.success? && tender.token.blank?
+    tender.token =
+      transaction.response.params["braintree_transaction"]["credit_card_details"]["token"]
+    tender.save!
+  end
+end
+```
+
```

Comment:
One of the most common decorations in the operation classes is to decorate  #transaction_options. This method exists to pass customized data to the active merchant implementation.  I think its notable enough to mention here.

Reply:
added it